CDCgov/prime-reportstream
# Engagement/angela/11780/bugfix sent files

This PR ...

The actual problem that was causing the failures in the linked ticket was that la-doh was having files generated with the same name for the elr and elr-secondary receiver. After sending, files were being uploaded to the blobstore in none/send. This PR changes the location files are uploaded to the blobstore to none/receiver.fullname which will put the files with the same name for different receivers into different folders. Added an EventAction.OTHER to be used as a default and changed the default action on BlobAccess.uploadBody() so that if it's called with no action for some reason, those files will be separated from files that have completed the pipeline. I also successfully built without the action to confirm that there was nowhere currently calling uploadBody() using the default action. The added test successfully failed in master and passes on this branch.

Test Steps:
1. run the test that was added
2. verify it passes
3. copy the test
4. check out master
5. paste the test into the appropriate file in master
6. run the test
7. verify it fails
8. (revert your code)
9. switch back to the branch
10. remove the default action from BlobAccess.uploadBody() and confirm that it still builds (I believe there may be one test somewhere that uses the default, so you may need to temporarily change that test also)
11. Unsure that manual testing can be performed locally.

## Changes
- Changed ActionHistory.trackSentReport() to call BlobAccess.uploadBody() with a subfolderName argument of receiver.fullName instead of "send".
- Changed the default action parameter in BlobAccess.uploadBody() from EventAction.NONE (which is what is passed by ActionHistory.trackSentReport()) to EventAction.OTHER.
- Added EventAction.OTHER and its related TaskAction.
- Added EventAction.OTHER to WorkflowEngine.finishedField() to be similar to EventAction.NONE. The differentiating point is that NONE is for when actions have been completed and no further action is needed and OTHER is meant to be used as a default in case something goes wrong.
- Added migration file to create the OTHER TaskAction. TaskAction is autogenerated by jooq.
- Updated ActionHistoryTests.test trackSentReport() to mock BlobAccess.uploadBody() using EventAction.NONE because that's what ActionHistory.TrackSentReport() is using.
- Added ActionHistoryTests.test trackSentReportSameFilenameDifferentReceiver() to test that the change to ActionHistory.trackSentReport() would successfully result in different blob urls for a file with the same name going to two receivers.
- Updated BlobAccessTests.upload body() to reflect the change to the default action in BlobAccess.uploadBody()

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #11780
